tools/go_generics/generics_tests/interface/output/output.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2019 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+type FooNew struct {
+	Q
+	Bar map[string]Q `json:"bar,omitempty"`
+}
+
+type BazNew struct {
+	T someTypeNotT
```

Comment:
@amscanne it actually doesn't break !

Reply:
I think that this staying the same is the expected behavior, the fact that all the _selector_ expressions are rewritten causes the conflict. The question is whether or not the following happens:

b := BazNew{}
b.T = 0 // will be rewritten to b.Q

Or even this:

b := otherpkg.UnrelatedType{}
b.T = 0 // Will be rewritten.

Sorry for the extended back and forth. I’m probably not being as clear as I could be. The issue is somewhat subtle though, but would be very difficult to diagnose.